### ISSUE TITLE
Missing requirement [823.0] osgi.wiring.package;

### ISSUE THREAD

**Issue by thibaultdelor** [Adobe-Consulting-Services/acs-aem-commons#533]
Our server seems to not be able to restart because of ACS. The error is the following : 
```
[FelixDispatchQueue] com.adobe.acs.acs-aem-commons-bundle FrameworkEvent ERROR (org.osgi.framework.BundleException: Unresolved constraint in bundle com.adobe.acs.acs-aem-commons-bundle [823]: Unable to resolve 823.0: missing requirement [823.0] osgi.wiring.package; (&(osgi.wiring.package=org.scribe.builder.api)(version>=1.3.0)(!(version>=2.0.0))))
org.osgi.framework.BundleException: Unresolved constraint in bundle com.adobe.acs.acs-aem-commons-bundle [823]: Unable to resolve 823.0: missing requirement [823.0] osgi.wiring.package; (&(osgi.wiring.package=org.scribe.builder.api)(version>=1.3.0)(!(version>=2.0.0)))
        at org.apache.felix.framework.Felix.resolveBundleRevision(Felix.java:4095)
        at org.apache.felix.framework.Felix.startBundle(Felix.java:2114)
        at org.apache.felix.framework.Felix.setActiveStartLevel(Felix.java:1368)
        at org.apache.felix.framework.FrameworkStartLevelImpl.run(FrameworkStartLevelImpl.java:308)
        at java.lang.Thread.run(Thread.java:745)
```

When we deploy manually ACS, everything is OK. When we deploying using maven it works the first but it seems that deploying several time is a problem. The server won't be failing until we restart.
We haven't manage to reproduce in a consistent manner, it's failing randomly after ~1 week of active deployments. On one of our AEM instance we have excluded ACS from maven and just deployed ACS manually once, this instance has been stable since then (1+ month).

We are not doing anything fancy, we are not messing with the package manager, we are just deploying our app with maven few times a day.

**Comment by davidjgonzalez** [comment]
What version of AEM are you using? 

What is the output of searching for org.scribe.builder.api via depfinder (http://localhost:4502/system/console/depfinder) on the offending box?

Does this only happen on 1 environment, or consistently across all environments w ACS commons installed?

**Comment by thibaultdelor** [comment]
AEM 6.1 with ACS 1.10.2
That use to happen on both our environments. Now that we took out ACS from maven on one of this environment, it's not doing it anymore on this one.
Our failing environment is currently broken so most of the tools aren't working, I'll try to get the output of depfinder ASAP

**Comment by davidjgonzalez** [comment]
Also - can you check to ensure that the `com.adobe.granite.scribe` bundle is started (/system/console/bundles) and that the version is >= 1.3.0 and < 2.0.0

**Comment by thibaultdelor** [comment]
Here is the depfinder ouptut :
![screen_shot_2015-09-14_at_1 09 19_pm](https://cloud.githubusercontent.com/assets/693797/9841345/f2f2f2b8-5ae1-11e5-9d83-4be249246285.png)
In package manager on the broken environment, `com.adobe.granite.scribe` is at version 1.3.8 and active.

**Comment by davidjgonzalez** [comment]
@thibaultdelor can you also goto /system/console/bundles, click on the ACS AEM Commons bundle (which should not be active due to this unresolved dependency) and take a screenshot of the expanded bundle, and specifically anything in red?

**Comment by thibaultdelor** [comment]
I could take the screenshot but there is nothing in red. The only red bits I could find in any of our bundles is on the ACS twitter bundle but that's fine. Because of the failing restart few of our bundles are not active but installed :
![screen shot 2015-09-14 at 1 58 50 pm](https://cloud.githubusercontent.com/assets/693797/9841873/3ecb77a8-5ae9-11e5-8b6f-5f9c124b21b9.png)

None of them have missing dependencies. Clicking on the start button for ACS doesn't do anything but writing to the error log a huge amount of statements, starting by the error on my first post.

**Comment by yupadhyay** [comment]
I get error with twitter package in AEM 6.1
![screen shot 2015-09-14 at 11 13 25 am](https://cloud.githubusercontent.com/assets/3046343/9857852/fa245096-5ad1-11e5-9c6a-82accc97469b.png)

**Comment by yupadhyay** [comment]
Twitter package get resolved after installing https://github.com/Adobe-Consulting-Services/com.adobe.acs.bundles.twitter4j I think this should be part of ACS common package as well if not already.

**Comment by justinedelson** [comment]
@thibaultdelor - try 1.10.4 (or later). This was fixed with #494.

@yupadhyay Twitter is a totally different issue and unrelated to this problem.

**Comment by thibaultdelor** [comment]
@justinedelson Interesting. The thing is that `com.adobe.granite.scribe` is at version `1.3.8`.
The problem seem deeper than that, all our custom osgi packages refuse to start. It's like the server is corrupted. For example the jackson library refuse to start. If I delete it and reinstall it, then it works. The more I investigate, the less I suspect ACS to be the cause.
Thanks for your help! You have all been very helpful.

**Comment by justinedelson** [comment]
I do agree that #494 was a weird issue and I wasn't thrilled by the change I made to fix that. But it did seem to resolve that particular user's issue and I haven't heard of it reoccurring since 1.10.4 (which, granted, wasn't that long ago). I would agree that an issue with other bundles implies this is a bigger issue.

**Comment by boser87** [comment]
I still get this issue:

org.osgi.framework.BundleException: Unresolved constraint in bundle com.adobe.acs.acs-aem-commons-bundle [473]: Unable to resolve 473.0: missing requirement [473.0] osgi.wiring.package; (&(osgi.wiring.package=org.scribe.builder.api)(version>=1.3.0)(!(version>=2.0.0)))

Could you please advice on how to fix this?

Adobe Experience Manager, Version 6.1.0
acs-aem-commons-content-1.10.2.zip

The issue happens everytime I restart AEM.

**Comment by justinedelson** [comment]
@boser87 take a look at #681. In retrospect, that was the underlying issue and #494 should not have been implemented.

**Comment by KiranVedantam** [comment]
Hi i have faced the similar issue and the below steps solved it.

System config:
AEM Version: 6.1
ACS-Commons package: 2.8.0

Issue:
 *ERROR* [FelixDispatchQueue] com.adobe.acs.acs-aem-commons-bundle-twitter FrameworkEvent ERROR (org.osgi.framework.BundleException: Unresolved constraint in bundle com.adobe.acs.acs-aem-commons-bundle-twitter [448]: Unable to resolve 448.0: missing requirement [448.0] osgi.wiring.package; (&(osgi.wiring.package=twitter4j)(version>=3.0.5)(!(version>=4.0.0))))

You will also see the twitter bundle inactive(As shown in the screenshot by @yupadhyay )

steps:
As said in the link (https://adobe-consulting-services.github.io/acs-aem-commons/features/twitter.html)
1. Start CQ Configuration Manager component, which is disabled by default (for 6.1+ only)
2. Download the content package with an OSGi bundle of this library (from https://github.com/Adobe-Consulting-Services/com.adobe.acs.bundles.twitter4j/releases/tag/com.adobe.acs.bundles.twitter4j-1.0.0) and install it in the package manager.
3. Restart the server (If you need a fresh log without error, i did it as it is my local)
4. You wont see the error again.
Thanks.

**Comment by Aourin** [comment]
10.02.2017 12:33:08.955 *ERROR* [FelixStartLevel] ERROR: Error starting jcrinstall:/apps/acs-commons/install/acs-aem-commons-bundle-twitter-2.9.0.jar (org.osgi.framework.BundleException: Unresolved constraint in bundle com.adobe.acs.acs-aem-commons-bundle-twitter [918]: Unable to resolve 918.0: missing requirement [918.0] osgi.wiring.package; (&(osgi.wiring.package=twitter4j)(version>=3.0.5)(!(version>=4.0.0))))
org.osgi.framework.BundleException: Unresolved constraint in bundle com.adobe.acs.acs-aem-commons-bundle-twitter [918]: Unable to resolve 918.0: missing requirement [918.0] osgi.wiring.package; (&(osgi.wiring.package=twitter4j)(version>=3.0.5)(!(version>=4.0.0)))

Just added ACS as per the instructions in one of the feature tutorials http://adobe-consulting-services.github.io/acs-aem-commons/features/versioned-clientlibs.html

**Comment by davidjgonzalez** [comment]
@Aourin 

1) Are you using the ACS Commons twitter functionality? If not, the i recommend the "min" package
2) If you are using the ACS Commons twitter functionality, you must make sure you install the ACS Commons Twittter bundle AND the Twitter4J OSGi bundle (check the ACS Clmmons Twitter feature page)

TL;DR use the `min` package if you arent using ACS Commons Twitter feature.

**Comment by sasconsul** [comment]
FYI: the page defining the dependency to alter is https://adobe-consulting-services.github.io/acs-aem-commons/pages/thirdparty.html
